src/validators/SignedKeyRequestValidator.sol

```diff
@@ -0,0 +1,135 @@
+     *
+     *  @param requestingFid The fid of the entity requesting to add
+     *                       a signer key.
+     *  @param requestSigner Signer address. Must be the owner of
+     *                       the requestingFid fid.
+     *  @param signature     EIP-712 SignedKeyRequest signature.
+     *  @param deadline      block.timestamp after which signature expires.
+     */
+    struct SignedKeyRequest {
+        uint256 requestFid;
+        address requestSigner;
+        bytes signature;
+        uint256 deadline;
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                                 EVENTS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @dev Emit an event when the admin sets a new IdRegistry contract address.
+     *
+     * @param oldIdRegistry The previous IdRegistry address.
+     * @param newIdRegistry The new IdRegistry address.
+     */
+    event SetIdRegistry(address oldIdRegistry, address newIdRegistry);
+
+    /*//////////////////////////////////////////////////////////////
+                              CONSTANTS
+    //////////////////////////////////////////////////////////////*/
+
+    bytes32 internal constant _METADATA_TYPEHASH =
+        keccak256("SignedKeyRequest(uint256 requestFid,bytes key,uint256 deadline)");
+
+    /*//////////////////////////////////////////////////////////////
+                                STORAGE
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @dev The IdRegistry contract.
+     */
+    IdRegistryLike public idRegistry;
+
+    /*//////////////////////////////////////////////////////////////
+                               CONSTRUCTOR
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Set the IdRegistry and owner.
+     *
+     * @param _idRegistry   IdRegistry contract address.
+     * @param _initialOwner Initial contract owner address.
+     */
+    constructor(address _idRegistry, address _initialOwner) EIP712("Farcaster MetadataValidator", "1") {
+        idRegistry = IdRegistryLike(_idRegistry);
+        _transferOwnership(_initialOwner);
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                               VALIDATION
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Validate the SignedKeyRequest metadata associated with a signer key.
+     *         (Key type 1, Metadata type 1)
+     *
+     * @param key                   The EdDSA public key of the signer.
+     * @param signedKeyRequestBytes An abi-encoded SignedKeyRequest struct, provided as the
+     *                              metadata argument to KeyRegistry.add.
+     *
+     * @return true if signature is valid and signer owns requestFid, false otherwise.
+     */
+    function validate(
+        uint256, /* userFid */
+        bytes memory key,
```

Comment:
should we also validate that the length of the key is == 32 bytes? 

this prevents people from making further mistakes when generating keys.

Reply:
Yeah, that's a good idea and an easy check.